### PR TITLE
users endpoint version 3

### PR DESCRIPTION
**Pull request by sahils-tenable**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Note** Will update the structure from tio.v3.vm.users() to tio.v3.user() soon.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Python Version(s) Tested:
* Tenable.sc version (if necessary):

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by tomeara — CHANGES_REQUESTED** [review]

File: tests/io/v3/vm/test_users.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                          "f3cd0bb2-cabb-4825-9d0c-49c77fe5fba7",
+                          "00000000-0000-0000-0000-000000000000",
+                          "2d5c70da-b177-43ed-8325-a25a846c8977",
+                          "a507f383-3130-4c89-b202-b69ad9a75a84",
+                          "afed07ce-8e51-4574-a420-90057fea6a7f",
+                      ],
+                      "lockout": 0,
+                      "container_uuid": "270f77d7-3b5b-478c-ac06-be827c00753e",
+                      "lastlogin": 1605630009020,
+                      "uuid_id": "73cc516b-51d0-445a-8e0d-6f618455770e",
```
[the remark this answers is not in the document]
Comment (sahils-tenable) [reply tests/io/v3/vm/test_users.py:44]: uuid_id => is being described as uuid for the user. 

--> https://developer.tenable.com/reference#users-details

File: tenable/io/v3/vm/users.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+            :obj:`None`:
+                The verification code was valid and two-factor is enabled.
+
+        Examples:
+            >>> tio.v3.vm.users.verify_two_factor(1, 'abc123')
+        '''
+        self._post(f'{user_id}/two-factor/verify-code', json={
+                'verification_code': code})
+
+    def impersonate(self, name: str) -> None:
```
Comment (tomeara) [inline tenable/io/v3/vm/users.py:264]: We should leave this out

File: tests/io/v3/vm/test_users.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                      "login_fail_count": 0,
+                      "login_fail_total": 3,
+                      "enabled": True,
+                      "undeletable": False,
+                      "two_factor": {
+                          "sms_phone": "+15551236789",
+                          "sms_enabled": 1,
+                          "email_enabled": 0,
+                      },
+                      "group_uuids": [
```
Comment (tomeara) [inline tests/io/v3/vm/test_users.py:81]: Let's go with `group_ids`

File: tests/io/v3/vm/test_users.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                          "f3cd0bb2-cabb-4825-9d0c-49c77fe5fba7",
+                          "00000000-0000-0000-0000-000000000000",
+                          "2d5c70da-b177-43ed-8325-a25a846c8977",
+                          "a507f383-3130-4c89-b202-b69ad9a75a84",
+                          "afed07ce-8e51-4574-a420-90057fea6a7f",
+                      ],
+                      "lockout": 0,
+                      "container_uuid": "270f77d7-3b5b-478c-ac06-be827c00753e",
+                      "last_login": 1605630009020,
+                      "uuid_id": "73cc516b-51d0-445a-8e0d-6f618455770e"
```
Comment (tomeara) [inline tests/io/v3/vm/test_users.py:91]: we should either rename this, or leave it out

**Review by tomeara — CHANGES_REQUESTED** [review]
Need to remove one of the calls. Added some more comments

**Review by tomeara — APPROVED** [review]
Looks good to me!

**Pull request merged**